examples/feature_selection/plot_permutation_test_for_classification.py

```diff
@@ -22,48 +20,115 @@
+# no dependency between features and labels. The
+# :class:`~sklearn.svm.SVC` classifier and :ref:`accuracy_score` are used.
+#
+# :func:`~sklearn.model_selection.permutation_test_score` generates a null
+# distribution by calculating the accuracy of the classifier
+# on 1000 different permutations of the dataset, where features
+# remain the same but labels undergo different permutations. This is the
+# distribution for the null hypothesis which states there is no dependency
+# between the features and labels. An empirical p value is then calculated as
+# the percentage of permutations for which the score obtained is greater
+# that the score obtained using the original data.
 
-svm = SVC(kernel='linear')
+clf = SVC(kernel='linear', random_state=7)
 cv = StratifiedKFold(2)
 
-score, permutation_scores, pvalue = permutation_test_score(
-    svm, X, y, scoring="accuracy", cv=cv, n_permutations=100, n_jobs=1)
-
-print("Classification score %s (pvalue : %s)" % (score, pvalue))
-
-# #############################################################################
-# View histogram of permutation scores
-plt.hist(permutation_scores, 20, label='Permutation scores',
-         edgecolor='black')
-ylim = plt.ylim()
-# BUG: vlines(..., linestyle='--') fails on older versions of matplotlib
-# plt.vlines(score, ylim[0], ylim[1], linestyle='--',
-#          color='g', linewidth=3, label='Classification Score'
-#          ' (pvalue %s)' % pvalue)
-# plt.vlines(1.0 / n_classes, ylim[0], ylim[1], linestyle='--',
-#          color='k', linewidth=3, label='Luck')
-plt.plot(2 * [score], ylim, '--g', linewidth=3,
-         label='Classification Score'
-         ' (pvalue %s)' % pvalue)
-plt.plot(2 * [1. / n_classes], ylim, '--k', linewidth=3, label='Luck')
-
-plt.ylim(ylim)
-plt.legend()
-plt.xlabel('Score')
+score_iris, perm_scores_iris, pvalue_iris = permutation_test_score(
+    clf, X, y, scoring="accuracy", cv=cv, n_permutations=1000)
+
+score_rand, perm_scores_rand, pvalue_rand = permutation_test_score(
+    clf, X_rand, y, scoring="accuracy", cv=cv, n_permutations=1000)
+
+# %%
+# Original data
+# ^^^^^^^^^^^^^
+#
+# Below we plot a histogram of the permutation scores (the null
+# distribution). The red line indicates the score obtained by the classifier
+# on the original data. The score is much better than those obtained by
+# using permuted data and the p value is thus very low. This indicates that
+# there is a low likelihood that this good score would be obtained by chance
+# alone. It provides evidence that the iris dataset contains real dependency
+# between features and labels and the classifier was able to utilize this
+# to obtain good results.
+
+fig, ax = plt.subplots()
+
+ax.hist(perm_scores_iris, bins=20)
+ax.axvline(score_iris, ls='--', color='r')
+score_label = (f"Score on original\ndata: {score_iris:.2f}\n"
+               f"(p value: {pvalue_iris:.3f})")
+ax.text(0.7, 260, score_label, fontsize=12)
```

Comment:
It could be good to have `xlabel` and `ylabel` on the figure.

```suggestion
ax.text(0.7, 260, score_label, fontsize=12)
ax.set_xlabel("Accuracy score")
ax.set_ylabel("Probability (or count)") # depending of ax.hist
```